### DISCUSSION TITLE
Confused with how to uncap TPS.

### DISCUSSION THREAD

**Discussion by Qualzz** [hajimehoshi/ebiten#2623] [Q&A] [answered]
Hello, 
I'm trying to do a turbo mode. Basically if the turbo mode is off, then the Ticks Per Second ( updates if i'm right ) should be 60 ( in my case )

If the turbo mode is on, i want the updates to run as fast as possible that the cpu could allow, while not changing the FPS. 

But i'm a bit confused about all the variables, the deprecated ones etc. 

For exemple, the "UncappedTPS" will just sync with the FPS as mentioned in the doc, and is now replaced with SyncWithFPS, while the name suggest that it's was **not** capped. 

Is there any way to have a turbo mode without setting manually the `ebiten.SetTPS() ` to an absurd value ?

**Comment by hajimehoshi** [comment]
I'm not sure what is the turbo mode in your game. Assuming that is just a high speed, I would set with a high TPS like 180. Doesn't this work to you?

**Comment by Qualzz** [comment, marked as answer]
Not necessary, basically i'm doing some genetic selection of the best performant objects in a simulation. 

So i want to be able to have the generations to run as fast as possible with the turbo mode. 

But i think i'll just implement fast forward options with different speed levels ( x2, x4, x8, x16 etc ) so i'll have a better control.

**Reply by hajimehoshi** [reply]
So you can adjust TPS like so.

**Reply by hajimehoshi** [reply]
> If the turbo mode is on, i want the updates to run as fast as possible that the cpu could allow, while not changing the FPS.

I think I understand what you want. Ebitengine doesn't provide such option so I would have an independent goroutine to run the logic with much less sleeps

**Reply by Qualzz** [reply]
Yeah it's fine like this, btw thank you for this amazing library, while i'm not a go expert, with the help of chat GPT and your lib, i've made in only a few hours a great prototype, with my tanks running around, spinning, shooting at each others, with debug infos, healthbars, sensor states, genetics mutations  and cross over ( which are also visible by changing the colors of the offsprings), fast forward neural network etc and all I need, really impressed, i think this is not the last time i'll use it !
